src/trip_state.py

```diff
@@ -104,12 +104,41 @@ def set_trip_state(self, trip_id: str, trip_state: TripState) -> None:
 
     @tracer.wrap()
     def get_trip_state(self, trip_id: str) -> Optional[TripState]:
-        self._purge_trips_state_if_overnight()
+        self._cleanup_trip_states()
         trip = self.trips.get(trip_id)
         if trip:
             return {**trip}
         return None
 
+    def _cleanup_trip_states(self) -> None:
+        """
+        Clean up trip states to prevent memory/CPU issues.
+        We don't care about yesterday's trip states, or ones older than 5 hours.
+        """
+        self._cleanup_stale_trip_states()
+        self._purge_trips_state_if_overnight()
+
+    @tracer.wrap()
+    def _cleanup_stale_trip_states(self, max_age_hours: int = 5) -> None:
```

Comment:
5 hours is probably not aggressive enough, but I don't want to accidentally drop trip state we need